recipes/matio/all/conanfile.py

```diff
@@ -0,0 +1,99 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class MatioConan(ConanFile):
+    name = "matio"
+    license = "BSD-2-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://sourceforge.net/projects/matio/"
+    description = "Matio is a C library for reading and writing binary MATLAB MAT files."
+    topics = ("conan", "matlab", "mat-file", "file-format", "hdf5")
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+    exports_sources = "CMakeLists.txt", "patches/*"
+    options = {
+        "shared": [True, False],
+        "extended_sparse": [True, False],
+        "fPIC": [True, False],
+        "mat73": [True, False],
+        "with_hdf5": [True, False],
+        "with_zlib": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "extended_sparse": True,
+        "fPIC": True,
+        "mat73": True,
+        "with_hdf5": True,
+        "with_zlib": True,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def requirements(self):
+        if self.options.get_safe("with_hdf5") not in (None, "None"):
+            self.requires("hdf5/[>=1.8 <1.13]")
+        if self.options.get_safe("with_zlib") not in (None, "None"):
+            self.requires("zlib/[>=1.2.3]")
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def config_options(self):
+        if self.settings.os == "Windows" or self.options.get_safe("shared"):
```

Comment:
The value of `shared` is not yet available in `config_options`.
```suggestion
        if self.settings.os == "Windows":
```